src/components/QuestionView.jsx

```diff
@@ -1,72 +1,132 @@
+  };
+
+  const calculateScore = () => {
+
+    // console.log("score")
+    // return Object.keys(correctAns).reduce((acc, questionId) => {
+    //     JSON.stringify(correctAns[questionId]) === JSON.stringify(givenAns[questionId]) && acc++
+    //     return acc
+    // }, 0)  
+    
+    return Object.keys(givenAns).reduce((acc, questionId) => {
+      let flag=true
+      console.log(givenAns)
+      Object.entries(givenAns[questionId]).forEach(([key,value])=>{
+        //console.log(key)
+        //console.log(correctAns[questionId][key])
+         
+          if (correctAns[questionId][key] !== givenAns[questionId][key]){
+              flag =false   
+          }    
+      })
+      if(flag){
+          acc++
+      }
+      return acc;
+      },0)
+
+}
+
+  const renderOptions = (flag) => {
+    var option_status =
+      flag == 0
+        ? defaultFormData
+        : givenAns[question.id];
+
+    return Object.entries(question.answers).map(([key, value]) => {
+    
+      let isChecked = option_status[key + "_correct"] === "true" ? true : false;
+      return (
+        value != null && (
+          <div className="form-check">
+            <input
+              key={key + "_" + question.id}
+              type="checkbox"
+              name={key + "_correct"}
+              value={key}
+              defaultChecked={isChecked}
+              onChange={(e) => handleChange(e)}
+            ></input>
+            <label>{value}</label>
+          </div>
+        )
+      );
+    });
+  };
+
+  const storeAnswer = (options) => {
+    //localStorage.setItem(question.id, JSON.stringify(options));
+    setGivenAns({...givenAns,[question.id]:options})
+    console.log({...givenAns,[question.id]:options});
+
   };
 
   return index < limit ? (
     <>
-      {skip == 1 ? <p>No Skippin'</p> : <p>Go on baby!</p>}
+      {/* {skip == 1 ? <p>No Skippin'</p> : <p>Go on baby!</p>} */}
       <h3>{question.question}</h3>
       <ul>
-        {Object.entries(question.answers).map(([key, value]) => {
-          return (
-            value != null && (
-              <div className="form-check">
-                <input
-                  key={key + "_" + question.id}
-                  type="checkbox"
-                  name={key + "_correct"}
-                  value={key}
-                  onChange={(e) => handleChange(e)}
-                ></input>
-                <label>{value}</label>
-              </div>
-            )
-          );
-        })}
+        {givenAns[question.id] == null || givenAns[question.id] == undefined
+          ? renderOptions(0)
+          : renderOptions(1)}
       </ul>
 
-      {/* {index > 0 && (
+      {index > 0 && (
         <button
           onClick={() => {
-            setIndex(index - 1);
-            setFormData(defaultFormData);
+            // JSON.stringify(formData) === JSON.stringify(defaultFormData)
+            //   ? setSkip(1)
+            //   : prev();
+            prev();
```

Comment:
If only one function is invoked then write it in a single line. No need to use {}.